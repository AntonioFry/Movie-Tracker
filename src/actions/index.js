export const showMovies = (movies) => ({
  type: 'SHOW_MOVIES',
  movies
});