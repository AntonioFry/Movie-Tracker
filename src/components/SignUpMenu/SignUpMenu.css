.sign-up-bar {
  background-color: red;
}

