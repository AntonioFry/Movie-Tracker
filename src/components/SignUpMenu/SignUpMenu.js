import React, { Component } from 'react';
import { SignUp } from '../SignUp/SignUp'
import { login, showError } from '../../actions';
import { connect } from 'react-redux';
import { getUser, addUser } from '../../api/apiCalls'
import { Redirect } from 'react-router'
import './SignUpMenu.css'

class SignUpMenu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email: '',
      password: '',
      display: '',
      error: null
    };
  }

  handleAddChange = (e) => {
    const {name, value} = e.target
    this.setState({ [name]: value })
  }

  handleAdd = async (e) => {
    try {
      e.preventDefault()
      const response = await addUser(this.state);
      console.log(response)
      if (response === true) {
        this.props.login(this.state)
        this.setState({display: 'loggedIn'})
      } else {
        this.setState({ error: "Email has already been used" });
      }
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  // addUser = async (user) => {
  //   try {
  //     const options = {
  //       method: 'POST',
  //       body: JSON.stringify(user),
  //       headers: {
  //         'Content-Type': 'application/json'
  //       }
  //     }
      
  //     const response = await fetch('http://localhost:3000/api/users/new', options)
  //     const result = await response.json()
      
  //     if (!response.ok) {
  //       return this.props.showError('This user already exists')
  //     } else {
  //       return getUser(this.state)
  //     }

  //   } catch (error) {
  //     console.log(error.message);
  //   }
  // }

  render() {
    const isLoggedIn = this.state.display === 'loggedIn'
    let view;

    if(!isLoggedIn) {
      view = <SignUp name={this.state.name} email={this.state.email} password={this.state.password} handleAddChange={this.handleAddChange} handleAdd={this.handleAdd}/>
    } else {
      view = <Redirect to='/LoggedIn'/>
    }
    return (
      <section className='sign-up-bar' >
        {view}
        {this.state.error && <h2 className='error'>{this.state.error}</h2>}
      </section>
    )
  }
}

const mapStateToProps = (store) => ({
  showError: store.showError
});


const mapDispatchToProps = (dispatch) => ({
  login: (user) => dispatch(login(user)),
  showError: (error) => dispatch(showError(error))
});

export default connect(mapStateToProps, mapDispatchToProps)(SignUpMenu)
