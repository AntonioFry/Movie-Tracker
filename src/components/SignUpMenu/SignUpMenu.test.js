import React from 'react';
import { shallow } from 'enzyme';
import { SignUpMenu, mapStateToProps, mapDispatchToProps} from './SignUpMenu'
import { showError, login } from '../../actions'

describe('SignUpMenu', () => {
  let wrapper;
  let mockLogin;
  let mockShowErrorFunc;
  let mockShowError;
  let mockUser;

  beforeEach(() => {
    mockUser = {}
    mockLogin = jest.fn();
    mockShowErrorFunc = jest.fn()
    mockShowError = 'email incorrect';
    wrapper = shallow(<SignUpMenu showError={mockShowErrorFunc} login={mockLogin} showError={mockShowError} user={mockUser}/>);
  });

  describe('SignUpMenu Component', () => {
    it('should update state when handle add change is called', () => {
      const mockEvent = {target: {name: 'Nombre', value:'password'}}
      const expected = 'password';

      wrapper.instance().handleAddChange(mockEvent);

      expect(wrapper.state('Nombre')).toEqual(expected)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an error', () => {
      const mockState = {
        showError: 'wrong email and password',
        user:{}
      }
      const expected = {
        showError: 'wrong email and password'
      }
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with a login error', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = login(1, 'nombre@nombre.com', 'password')
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.login(1, 'nombre@nombre.com', 'password')

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})