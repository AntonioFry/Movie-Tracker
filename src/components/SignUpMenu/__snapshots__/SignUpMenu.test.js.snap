// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpMenu should match snapshot 1`] = `ShallowWrapper {}`;
